lib/network/service/event/tunnel.ex

```diff
@@ -0,0 +1,13 @@
+defmodule Helix.Network.Service.Event.Tunnel do
+
+  alias Helix.Network.Model.ConnectionClosedEvent
+  alias Helix.Network.Controller.Tunnel
+
+  def connection_closed(e = %ConnectionClosedEvent{}) do
+    tunnel = Tunnel.fetch(e.tunnel_id)
+
+    if Enum.empty?(Tunnel.get_connections(tunnel)) do
+      Tunnel.delete(tunnel)
+    end
```

Comment:
Note that this event handler might change in the future for 2 reasons

1) It might not be the best idea to immediately delete the tunnel after it gets empty because in the meantime another connection might be starting and if the tunnel is removed (before or after), we would have a race condition
2) It's overly expensive the return a formated list of structs representing connections only to know if one exists